### ISSUE TITLE
Compatibility with Spring Data Hopper

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-ldap#380]
Spring LDAP is currently incompatible with Spring Data Hopper. Compiling against Spring Data Commons 1.12.0.M1 produces four errors in `spring-ldap-core`:

```
/Users/awilkinson/dev/spring/spring-ldap/core/src/main/java/org/springframework/ldap/repository/query/LdapQueryMethod.java:42: error: constructor QueryMethod in class QueryMethod cannot be applied to given types;
        super(method, metadata);
        ^
  required: Method,RepositoryMetadata,ProjectionFactory
  found: Method,RepositoryMetadata
  reason: actual and formal argument lists differ in length
/Users/awilkinson/dev/spring/spring-ldap/core/src/main/java/org/springframework/ldap/repository/support/LdapRepositoryFactory.java:93: error: LdapRepositoryFactory.LdapQueryLookupStrategy is not abstract and does not override abstract method resolveQuery(Method,RepositoryMetadata,ProjectionFactory,NamedQueries) in QueryLookupStrategy
    private final class LdapQueryLookupStrategy implements QueryLookupStrategy {
                  ^
/Users/awilkinson/dev/spring/spring-ldap/core/src/main/java/org/springframework/ldap/repository/support/LdapRepositoryFactory.java:94: error: method does not override or implement a method from a supertype
        @Override
        ^
/Users/awilkinson/dev/spring/spring-ldap/core/src/main/java/org/springframework/ldap/repository/support/QueryDslLdapRepository.java:37: error: QueryDslLdapRepository is not abstract and does not override abstract method exists(Predicate) in QueryDslPredicateExecutor
public class QueryDslLdapRepository<T> extends SimpleLdapRepository<T> implements QueryDslPredicateExecutor<T> {
```

Testing the existing code against Spring Data Commons 1.12.0.M1 results in 14 test failures due to the unimplemented method that's been added to the `QueryLookupStrategy` interface:

```
Caused by: java.lang.AbstractMethodError: org.springframework.ldap.repository.support.LdapRepositoryFactory$LdapQueryLookupStrategy.resolveQuery(Ljava/lang/reflect/Method;Lorg/springframework/data/repository/core/RepositoryMetadata;Lorg/springframework/data/projection/ProjectionFactory;Lorg/springframework/data/repository/core/NamedQueries;)Lorg/springframework/data/repository/query/RepositoryQuery;
	at org.springframework.data.repository.core.support.RepositoryFactorySupport$QueryExecutorMethodInterceptor.<init>(RepositoryFactorySupport.java:435)
	at org.springframework.data.repository.core.support.RepositoryFactorySupport.getRepository(RepositoryFactorySupport.java:220)
	at org.springframework.data.repository.core.support.RepositoryFactoryBeanSupport.initAndReturn(RepositoryFactoryBeanSupport.java:266)
	at org.springframework.data.repository.core.support.RepositoryFactoryBeanSupport.afterPropertiesSet(RepositoryFactoryBeanSupport.java:252)
	at org.springframework.ldap.repository.support.LdapRepositoryFactoryBean.afterPropertiesSet(LdapRepositoryFactoryBean.java:47)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1637)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1574)
	... 58 more
```

/cc @olivergierke

**Comment by rwinch** [comment]
I think I will need to look to @olivergierke on how to best resolve this and remain passive with previous versions.

**Comment by rwinch** [comment]
Bump @olivergierke

**Comment by rwinch** [comment]
Speaking to the Spring Data team offline they pointed me to this change

https://github.com/spring-projects/spring-data-keyvalue/commit/d1df6d5e7681f54bdaa5fb1fa0d5d88331d9d6a0#diff-46bfee26d19fa777408c4077ee8ceb17R179

**Comment by rwinch** [comment]
This is now Fixed via #390
